File: examples/example3.py
#!/usr/bin/python
#
# File: example3.py       I. Henry   March 30 2005
#
# Python translation of example2.c from the WFDB Programmer's Guide
# 							   
# Copyright (C) 2005 Isaac C. Henry (ihenry@physionet.org)

import wfdb, sys

def main(argv):
    a = wfdb.WFDB_Anninfo()
    annot = wfdb.WFDB_Annotation()
    if len(argv) < 3:
        print "usage:", argv[0], "annotator record"
        sys.exit(1)
    a.name = argv[1]
    a.stat = wfdb.WFDB_READ
    wfdb.sampfreq(argv[2])
    if wfdb.annopen(argv[2], a, 1) < 0: sys.exit(2)
    while wfdb.getann(0, annot) == 0:
        if annot.aux is not None:
            aux = annot.aux[1:]
        else:
            aux = ""
        print wfdb.timstr(-annot.time), \
              "(" + str(annot.time)+ ")", \
              wfdb.annstr(annot.anntyp), \
              annot.subtyp, \
              annot.chan, \
              annot.num, \
              aux
    wfdb.wfdbquit()

if __name__ == "__main__":
    main(sys.argv)

File: examples/example3.pl
#!/usr/bin/perl
#
# File: example3.pl       I. Henry   March 29 2005
#
# Perl translation of example3.c from the WFDB Programmer's Guide
#						   
# Copyright (C) 2005 Isaac C. Henry (ihenry@physionet.org)

package wfdb;
use wfdb;

$a = new wfdb::WFDB_Anninfo();
$annot = new wfdb::WFDB_Annotation();
if ( @ARGV < 2 ) {
    print STDERR "usage: ", $0, " annotator record\n";
    exit(1);
} 
$a->{name} = $ARGV[0];
$a->{stat} = $WFDB_READ;
sampfreq($ARGV[1]);
exit(2) if (annopen($ARGV[1], $a, 1) < 0);
while (getann(0, $annot) == 0) {
    print timstr(-$annot->{time}), " (",
    $annot->{time}, ") ",
    annstr( $annot->{anntyp} ), " ",
    $annot->{subtyp}, " ",
    $annot->{chan}, " ",
    $annot->{num}, " ",
    substr($annot->{aux}, 1), "\n";
}
wfdbquit();
